### PR TITLE
feat(nuxi): reload nuxt when `restart` hook is called

### LINKED ISSUE #18678: Add hook to restart the nuxt server
I think it can be done right after this line with a `hookOnce('restart')`:
https://github.com/nuxt/nuxt/blob/434d36b532f564638d39029525d0b40d97bfe85f/packages/nuxi/src/commands/dev.ts#L78

This will let modules such as Tailwind to restart the server on config changes: 
https://github.com/nuxt-modules/tailwindcss/blob/12de7311aa1fe5f8877ad622a95278dc86303703/src/module.ts#L125-L126

### PR DESCRIPTION
**Pull request by Baroshem**
### 🔗 Linked issue

#18678 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Resolves #18678

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxi/src/commands/dev.ts [hunk 1]
```diff
@@ -76,6 +76,8 @@ export default defineNuxtCommand({
           await currentNuxt.close()
         }
         currentNuxt = await loadNuxt({ rootDir, dev: true, ready: false })
+        currentNuxt.hook('restart', () => load(true))
```
Comment (pi0) [inline packages/nuxi/src/commands/dev.ts:79]:
```suggestion
        currentNuxt.hookOnce('restart', () => load(true))
```

**Review by pi0 — APPROVED** [review]
💯

File: packages/nuxi/src/commands/dev.ts [hunk 2]
```diff
@@ -76,6 +76,8 @@ export default defineNuxtCommand({
           await currentNuxt.close()
         }
         currentNuxt = await loadNuxt({ rootDir, dev: true, ready: false })
+        currentNuxt.hookOnce('restart', () => load(true))
```
Comment (danielroe) [inline packages/nuxi/src/commands/dev.ts:79]:
```suggestion
        currentNuxt.hooks.hookOnce('restart', () => load(true))
```

File: packages/schema/src/types/hooks.ts [hunk 3]
```diff
@@ -73,6 +73,11 @@ export interface NuxtHooks {
    * @returns Promise
    */
   'close': (nuxt: Nuxt) => HookResult
+  /**
+   * Called when Nuxt instance is restarting.
+   * @returns Promise
+   */
+  'restart': () => HookResult
```
Comment (danielroe) [inline packages/schema/src/types/hooks.ts:80]:
```suggestion
  /**
   * Called to restart the current Nuxt instance.
   * @returns Promise
   */
  'restart': () => HookResult
```

File: packages/schema/src/types/hooks.ts [hunk 4]
```diff
@@ -73,6 +73,11 @@ export interface NuxtHooks {
    * @returns Promise
    */
   'close': (nuxt: Nuxt) => HookResult
+  /**
+   * Called when Nuxt instance is restarting.
```
Comment (danielroe) [inline packages/schema/src/types/hooks.ts:77]: Would you also add this to the docs? https://github.com/nuxt/nuxt/blob/b25b6c82c7fbd6dc3f3378c4bcac8bcc5ce361e5/docs/3.api/4.advanced/1.hooks.md#L32

File: docs/3.api/4.advanced/1.hooks.md [hunk 5]
```diff
@@ -38,6 +38,7 @@ Hook                     | Arguments                  | Description
 `kit:compatibility`      | `compatibility, issues`    | Allows extending compatibility checks.
 `ready`                  | `nuxt`                     | Called after Nuxt initialization, when the Nuxt instance is ready to work.
 `close`                  | `nuxt`                     | Called when Nuxt instance is gracefully closing.
+`restart`                | -                          | Called when Nuxt instance is restarting.
```
Comment (danielroe) [inline docs/3.api/4.advanced/1.hooks.md:41]:
```suggestion
`restart`                | -                          | Called to restart the current Nuxt instance.
```

**Review by danielroe — APPROVED** [review]
Nice! 👌 

(Deferring to v3.3 release as it's a feature.)

**Review by TheAlexLichter — APPROVED** [review]

File: docs/3.api/4.advanced/1.hooks.md [hunk 6]
```diff
@@ -38,6 +38,7 @@ Hook                     | Arguments                  | Description
 `kit:compatibility`      | `compatibility, issues`    | Allows extending compatibility checks.
 `ready`                  | `nuxt`                     | Called after Nuxt initialization, when the Nuxt instance is ready to work.
 `close`                  | `nuxt`                     | Called when Nuxt instance is gracefully closing.
+`restart`                | -                          | Called to restart the current Nuxt instance.
```
Comment (danielroe) [inline docs/3.api/4.advanced/1.hooks.md:41]:
```suggestion
`restart`                | -                          | Called to restart the current Nuxt instance. **This hook is currently only available on the [Edge Channel](/docs/guide/going-further/edge-channel/).** <!-- stabilityedge -->
```

**Pull request merged**